heroic-core/src/main/java/com/spotify/heroic/metric/LocalMetricManager.java

```diff
@@ -376,7 +374,7 @@ public QueryTrace buildTrace() {
             );
 
             // Add watcher to set to compute stats across all queries
-            quotaWatchers.add(quotaWatcher);
+            quotaWatchers.put(quotaWatcher.hashCode(), quotaWatcher);
```

Comment:
So about hashCode... This *should* be fine but the implementation is JVM dependent. It is also **not** simply driven by the the memory address since objects move during GC... 

So, I'd be happier if we'd defined hashCode but it will likely work as-is. Your call @malish8632 🙂

Reply:
I think the hashCode observation is important but as long as it works for this solution it's fine. Let's deploy and observe. This is pre-requisite for change to act on those stats - having this will help us understand the behavior better.

Reply:
👍🏻